source/features/comment-shortcuts.tsx

```diff
@@ -0,0 +1,97 @@
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {isEditable} from '../helpers/dom-utils';
+
+const shortcutClass = new Map<string, string>([
+	['e', '.rgh-edit-comment, [aria-label="Edit comment"]'],
+	['d', '[aria-label="Delete comment"]'],
+	['h', '[aria-label="Hide comment"]']
+]);
+
+function commentHash(comment: HTMLElement): string {
+	// What do I call this var??
+	const reviewComment = comment.closest('.js-comment[id^="pullrequestreview-"]');
+	const {hash} = reviewComment ? reviewComment.querySelector<HTMLAnchorElement>('a.js-timestamp')! : comment.querySelector<HTMLAnchorElement>('a.js-timestamp')!;
+
+	return reviewComment ? hash + '-body-html' : hash;
+}
+
+function runShortcuts(event: KeyboardEvent): void {
+	if (isEditable(event.target)) {
+		return;
+	}
+
+	if (['j', 'k'].includes(event.key)) {
+		event.preventDefault();
+
+		const items = select.all<HTMLAnchorElement>([
+			'.timeline-comment-group[id^="issue"]:not([href])', // Regular comments
+			'.timeline-comment-group[id^="pullrequestreview-"]:not([href])', // Base review comments (Approved/ChangesRequested)
+			'.review-comment.js-minimizable-comment-group' // Review comments
+		]).filter(element => !element.querySelector('.minimized-comment:not(.d-none)'));
+		// `j` goes to the next comment `k` goes back a comment
+		const direction = event.key === 'j' ? 1 : -1;
+		// Find current
+		const currentComment = select<HTMLAnchorElement>(':target')!;
+		const currentIndex = items.indexOf(currentComment);
+		// Nothing selected or were on the first comment
+		if (currentIndex + direction < 0 || currentIndex === -1) {
+			return;
+		}
+
+		// Find chosen
+		const chosen = items[Math.min(currentIndex + direction, items.length - 1)]; // Clamp it so it cant go past the last one
+
+		// Focus comment and dont put into history
+		location.replace(commentHash(chosen));
+		chosen.scrollIntoView();
+		return;
+	}
+
+	const actionClass = shortcutClass.get(event.key);
+	if (actionClass) {
+		event.preventDefault();
+		const target = select(':target');
+		if (!target || location.hash === '#partial-timeline') {
```

Comment:
This fails for any other `#hash`

The proper check is to verify that `:target` is a supported one like we do with the `indexOf` above. You might have to extract a `getItems` function so it it can be reused

Reply:
I removed this crazy hash thing. What a long day...